module-system/module-implementations/sov-nft-module/src/offchain.rs

```diff
@@ -0,0 +1,168 @@
+#[cfg(feature = "offchain")]
+use postgres::NoTls;
+use sov_modules_macros::offchain;
+
+#[cfg(feature = "offchain")]
+use crate::utils::get_collection_address;
+use crate::{Collection, CollectionAddress, Nft};
+
+// CREATE TABLE collection (
+//     collection_address TEXT PRIMARY KEY,
+//     collection_name TEXT NOT NULL,
+//     creator_address TEXT NOT NULL,
+//     frozen BOOLEAN NOT NULL,
+//     metadata_url TEXT,
+//     supply BIGINT NOT NULL
+// );
+
+#[offchain]
+pub fn track_collection<C: sov_modules_api::Context>(collection: &Collection<C>) {
+    // data extraction
+    let collection_name = collection.get_name();
+    let creator_address = collection.get_creator();
+    let frozen = collection.is_frozen();
+    let metadata_url = collection.get_collection_uri();
+    let supply = collection.get_supply();
+    let collection_address: CollectionAddress<C> =
+        get_collection_address(collection_name, creator_address.as_ref());
+    let collection_address_str = collection_address.to_string();
+    let creator_address_str = creator_address.to_string();
+    // postgres insert
+    tokio::task::block_in_place(|| {
+        if let Ok(conn_string) = std::env::var("POSTGRES_CONNECTION_STRING") {
+            match postgres::Client::connect(&conn_string, NoTls) {
+                Ok(mut client) => {
+                    let result = client.execute(
+                        "INSERT INTO collections (\
+                    collection_address, collection_name, creator_address,\
+                    frozen, metadata_url, supply)\
+                    VALUES ($1, $2, $3, $4, $5, $6)\
+                    ON CONFLICT (collection_address)\
+                    DO UPDATE SET collection_name = EXCLUDED.collection_name,\
+                                  creator_address = EXCLUDED.creator_address,\
+                                  frozen = EXCLUDED.frozen,\
+                                  metadata_url = EXCLUDED.metadata_url,\
+                                  supply = EXCLUDED.supply",
+                        &[
+                            &collection_address_str,
+                            &collection_name,
+                            &creator_address_str,
+                            &frozen,
+                            &metadata_url,
+                            &(supply as i64),
+                        ],
+                    );
+                    if let Err(e) = result {
+                        println!("Failed to execute query: {}", e);
+                    }
+                }
+                Err(e) => {
+                    println!("Failed to connect to the database: {}", e);
+                }
+            }
+        } else {
+            println!("Environment variable POSTGRES_CONNECTION_STRING is not set");
```

Comment:
nit: Can we `tracing::error!` instead of `print`ing in all this cases, or do we want to avoid a `tracing` dependency here?

Reply:
dependency should be fine! the more offchain or convenience libs and methods we can use, the more it shows the benefits of using the offchain component (in fact they can even add sentry logging etc and would not need to rely on emitting events or using on-chain compute). 
will add tracing in